Test/Tests/Integration/NodeManager.Tests.ps1

```diff
@@ -0,0 +1,250 @@
+
+    if($OurNode) {
+        return $true
+    }
+    return $false
+}
+
+function Test-ControllerReceivesNodeStatus {
+    Param (
+        [Parameter(Mandatory = $true)] [PSSessionT] $Session
+    )
+
+    # TODO
+
+    return $true
+}
+
+function Start-NodeMgr {
+    Param (
+        [Parameter(Mandatory = $true)] [PSSessionT] $Session
+    )
+
+    Invoke-Command -Session $Session -ScriptBlock {
+        [Diagnostics.CodeAnalysis.SuppressMessageAttribute(
+        "PSUseDeclaredVarsMoreThanAssignments", "",
+        Justification="Analyzer doesn't understand relation of Pester blocks"
+        )]
+        $NodeMgrJob = Start-Job -ScriptBlock {
+            contrail-nodemgr --nodetype contrail-vrouter
+        }
+    }
+}
+
+function Stop-NodeMgr {
+    Param (
+        [Parameter(Mandatory = $true)] [PSSessionT] $Session
+    )
+
+    Invoke-Command -Session $Session -ScriptBlock {
+        if(Get-Variable "NodeMgrJob" -ErrorAction SilentlyContinue) {
+            $NodeMgrJob | Stop-Job
+            Remove-Variable "NodeMgrJob"
+        }
+    }
+}
+
+function Install-Components {
+    Param (
+        [Parameter(Mandatory = $true)] [PSSessionT] $Session
+    )
+
+    Install-Extension -Session $Session
+    Install-DockerDriver -Session $Session
+    Install-Agent -Session $Session
+    Install-Nodemgr -Session $Session
+    New-NodeMgrConfig -Session $Session
+}
+
+function Uninstall-Components {
+    Param (
+        [Parameter(Mandatory = $true)] [PSSessionT] $Session
+    )
+
+    Uninstall-Nodemgr -Session $Session
+    Uninstall-Agent -Session $Session
+    Uninstall-DockerDriver -Session $Session
+    Uninstall-Extension -Session $Session
+}
+
+Describe "Node manager" {
+    It "starts" {
+        Eventually {
+            Test-NodeMgrLogs -Session $Session | Should Be True
+        } -Duration 60
+    }
+
+    It "connects to controller" {
+        Eventually {
+            Test-NodeMgrConnectionWithController -Session $Session | Should Be True
+        } -Duration 60
+    }
+
+    It "sets node state as 'Up'" -Pending {
+        Eventually {
+            Test-ControllerReceivesNodeStatus -Session $Session | Should Be True
+        } -Duration 60
+    }
+
+    BeforeEach {
+        Initialize-ComputeServices -Session $Session `
+            -SystemConfig $SystemConfig `
+            -OpenStackConfig $OpenStackConfig `
+            -ControllerConfig $ControllerConfig
+        Clear-NodeMgrLogs -Session $Session
+        Start-NodeMgr -Session $Session
+    }
+
+    AfterEach {
+        try {
+            Stop-NodeMgr -Session $Session
+            Clear-TestConfiguration -Session $Session -SystemConfig $SystemConfig
+        } finally {
+            Merge-Logs -LogSources (New-FileLogSource -Path (Get-ComputeLogsPath) -Sessions $Session)
+        }
+    }
+
+    BeforeAll {
+        $VMs = Read-TestbedsConfig -Path $TestenvConfFile
```

Comment:
We should consider **cooperative rebase** because of https://github.com/Juniper/contrail-windows-ci/pull/283/files#diff-b7aae0691fc57d2eb23c4ab71c4f58caR11

To be discussed...